src/LiveComponent/tests/Functional/Form/ComponentWithFormTest.php

```diff
@@ -143,4 +144,105 @@ public function testFormRemembersValidationFromInitialForm(): void
             ->assertContains('The title field should not be blank')
         ;
     }
+
+    public function testHandleCheckboxChanges(): void
+    {
+        /** @var LiveComponentHydrator $hydrator */
+        $hydrator = self::getContainer()->get('ux.live_component.component_hydrator');
+
+        /** @var ComponentFactory $factory */
+        $factory = self::getContainer()->get('ux.twig_component.component_factory');
+
+        /** @var FormComponent1 $component */
+        $component = $factory->create('form_component1');
+
+        $dehydrated = $hydrator->dehydrate($component);
+        $bareForm = [
+            'text' => '',
+            'textarea' => '',
+            'range' => '',
+            'choice' => '',
+            'choice_expanded' => '',
+            'choice_multiple' => ['2'],
+            'select_multiple' => ['2'],
+            'checkbox_checked' => '1',
+            'file' => '',
+            'hidden' => '',
+        ];
+        $fullBareData = array_merge(
+            $bareForm,
+            [
+                'checkbox' => null,
+            ]
+        );
```

Comment:
This extra variable is confusing. I believe it is because we want to ASSERT that the `checkbox` field equals null. But when we SEND the data to the server, we want this field to be omitted? Is that right?

If so, are you sure that's correct? With the new way of sending the data as JSON, if I have a form with an unchecked checkbox, the initial data sent to the frontend will store this value as `null` (and also, if I check and uncheck the box, the model will be set to `null`). And we DO now send `null` values to the Ajax calls.

Reply:
Fun fact is - in pure HTML world unchecked checkboxes would not be sent to the server at all and before parsing data as JSON live component was following those trails and it looked like perfect solution.

BUT since data is now sent as JSON and as Symfony Form component is handling nulls as unchecked state (as my updated test shows) I believe it will be simpler to just send all nulls.

This also simplifies this test case a lot by removing variable that probably only I knew why existed :)